prereise/gather/demanddata/nrel_efs/get_efs_data.py

```diff
@@ -288,6 +392,259 @@ def partition_by_sector(es, ta, year, sect=None, fpath="", save=False):
+
+    :param str es: An electrification scenario. Can choose one of: *'Reference'*,
+        *'Medium'*, or *'High'*.
+    :param str ta: A technology advancement. Can choose one of: *'Slow'*, *'Moderate'*,
+        or *'Rapid'*.
+    :param str flex: A flexibility scenario. Can choose one of: *'Base'* or
+        *'Enhanced'*.
+    :param int year: The selected year's worth of demand data. Can choose one of: 2018,
+        2020, 2024, 2030, 2040, or 2050.
+    :param set/list sect: The sectors for which .csv files are to be created. Can
+        choose any of: *'Transportation'*, *'Residential'*, *'Commercial'*,
+        *'Industrial'*, or *'All'*. Defaults to None.
+    :param str fpath: The file path where the demand data might be saved and to where
+        the sectoral data will be saved.
+    :param bool save: Determines whether or not the .csv file is saved. Defaults to
+        False. If the file is saved, it is saved to the same location as fpath.
+    :return: (*dict*) -- A dict of pandas.DataFrame objects that contain flexibility
+        data for each state and time step in the specified sectors.
+    :raises TypeError: if save is not input as a bool.
+    """
+
+    # Account for the immutable default parameters
+    if sect is None:
+        sect = {"All"}
+
+    # Check the inputs
+    es = _check_electrification_scenarios_for_partition(es)
+    ta = _check_technology_advancements_for_partition(ta)
+    flex = _check_flexibility_scenario(flex)
+    _check_year(year)
+    sect = _check_sectors(sect)
+    fpath = _check_path(fpath)
+    if not isinstance(save, bool):
+        raise TypeError("save must be input as a bool.")
+
+    # Specify the file name and path
+    csv_name = "EFSFlexLoadProfiles_" + es + ".csv"
+    csv_path = os.path.join(fpath, csv_name)
+
+    # Download the specified NREL EFS dataset if it is not already downloaded
+    if not os.path.isfile(csv_path):
+        download_flexibility_data({es}, fpath)
+
+    # Load the data from the downloaded .csv file as a DataFrame
+    df = pd.read_csv(csv_path)
+
+    # Trim the DataFrame for only the specified year
+    df = df.loc[df["Year"] == year]
+
+    # Trim the DataFrame for only the specified technology advancement
+    df = df.loc[df["TechnologyAdvancement"] == ta]
+
+    # Trim the DataFrame for only the specified flexibility scenario
+    df = df.loc[df["Flexibility"] == flex]
+
+    # Drop unnecessary "Year", "Electrification", and "TechnologyAdvancement" columns
+    df.drop(
+        columns=["Year", "Electrification", "TechnologyAdvancement", "Flexibility"],
+        inplace=True,
+    )
+
+    # Split the flexibility DataFrame by sector
+    sect_flex = {
+        i: df[df["Sector"] == i]
+        .drop(columns=["Sector"])
+        .groupby(["LocalHourID", "State"])
+        .sum()
+        .unstack()
+        for i in sect
+    }
+    sect_flex = {
+        i: sect_flex[i].set_axis(sorted(set(abv2state) - {"AK", "HI"}), axis="columns")
```

Comment:
Just to confirm the original order of the columns here matchers the new label order, i.e. `sorted(set(abv2state_- {"AK", "HI"})`?

Reply:
Yes. The reason we need to set the columns is because `unstack` creates double-valued columns (i.e., each column is a pair of "LoadMW" and the state). However, the pairs are already ordered according to the sorted states (if I remember correctly, I think `groupby` orders the columns by default). So, setting the columns in effect eliminates the additional "LoadMW" label on every column.

Reply:
I see. Then that's okay. I'm concerned about this simply because `set_axis` only change the column labels instead of moving corresponding data and if it happens to be different order, the table is messed up.

Reply:
Yeah, I agree. I was trying to find another way to get rid of the paired "LoadMW" labels, but this was the only way I could figure out. Since `groupby` orders the columns by default, we should be okay though.

Reply:
Are the labels a tuple, or a MultiIndex? Either way, we should be able to set them based on what's actually there, not what we expect. If it's a tuple, we can do a list comprehension; if it's a MultiIndex, we can use the `get_level_values` method.

Reply:
They are a MultiIndex. I can look into the `get_level_values` method that you suggest

Reply:
If after groupby, it gives something like`LoadMW_CA`, maybe doing following could be safer: `sect_flex[i].rename(lambda x:x[-2:], axis="columns", inplace=True)`

Reply:
After `groupby`, each column is essentially `('LoadMW', '[state]')`. It looks like `get_level_values` can be implemented easily by doing `sect_flex[i].columns.get_level_values(1)`

Reply:
Unless I'm missing something, it seems that I still need to use `set_axis` to keep this as a "one-line" piece of code: 

`sect_flex = {i: sect_flex[i].set_axis(sect_flex[i].columns.get_level_values("State"), axis="columns") for i in sect}`

However, this is safer than what is currently there because the use of `get_level_values` should generate column headers that are consistent with the order of the data.